### PR TITLE
allow using LWS without SOCKS5

### PR DESCRIPTION
**Pull request by underhood**
##### Summary
As a first step towards allowing usage of system LWS we allow building Netdata with LWS that was built without SOCKS5.
In case the user tries to use SOCKS5 we will log appropriate error into the logfile and ignore the setting.

More detailed discussion #8961 

##### Component Name
ACLK

##### Test Plan
Build with LWS without SOCKS 5 support. Should build without error. Example how to do that: Cripple installer.sh (remove `LWS_WITH_SOCKS5:bool=ON`) here (in else branch):
```
build_libwebsockets() {
  local env_cmd=''

  if [ -z "${DONT_SCRUB_CFLAGS_EVEN_THOUGH_IT_MAY_BREAK_THINGS}" ]; then
    env_cmd="env CFLAGS= CXXFLAGS= LDFLAGS="
  fi

  pushd "${1}" > /dev/null || exit 1
  if [ "$(uname)" = "Darwin" ] && [ -d /usr/local/opt/openssl ]; then
    run ${env_cmd} cmake \
      -D OPENSSL_ROOT_DIR=/usr/local/opt/openssl \
      -D OPENSSL_LIBRARIES=/usr/local/opt/openssl/lib \
      -D LWS_WITH_SOCKS5:bool=ON \
      $CMAKE_FLAGS \
      .
  else
    run ${env_cmd} cmake -D LWS_WITH_SOCKS5:bool=ON $CMAKE_FLAGS .
```
Attempting to use SOCKS5 proxy will result in error message and such setting to be ignored.

### CODE REVIEW

**Review by mfundul — DISMISSED** [review]

File: aclk/aclk_common.c [hunk 1]
```diff
@@ -144,14 +148,31 @@ const char *aclk_lws_wss_get_proxy_setting(ACLK_PROXY_TYPE *type)
...
             *type = PROXY_TYPE_HTTP;
         return proxy;
     }
 
     *type = aclk_verify_proxy(proxy);
+#ifndef LWS_WITH_SOCKS5
+    if (*type == PROXY_TYPE_SOCKS5) {
+        char *log = strdupz(proxy);
+        safe_log_proxy_censor(log);
+        fatal("Config var \"" ACLK_PROXY_CONFIG_VAR "\" set to use SOCKS5 proxy but this agent "
```
Comment (mfundul) [inline aclk/aclk_common.c:171]: We shouldn't kill the agent, but we should drop ACLK instead.

**Review by mfundul — APPROVED** [review]

File: aclk/aclk_common.c (hunk 1 above)
Comment (underhood) [reply aclk/aclk_common.c]: I changed it to behavior discussed on Slack.
Now proxy setting is checked at the beginning of the ACLK thread and will be disabled if sepcifically SOCKS5 is requested by config.

Comment (underhood) [reply aclk/aclk_common.c]: that means we will not `fatal` after last changes same as you suggested

**Review by mfundul — DISMISSED** [review]

**Comment by stelfrag** [comment]
I get 

```
ACLK_Main : socks_proxy environment variable set to use SOCKS5 proxy but Libwebsockets used doesn't have SOCKS5 support built in. Ignoring and checking for other options. Provided Value:"socks5://XXXXXXXXXXX@192.168.1.10:7000" (errno 2, No such file or directory)
```

But i do have it compiled

```
 LWS_WITH_RANGES = OFF
 LWS_PLAT_OPTEE = OFF
 LWS_WITH_ESP32 = OFF
 LWS_WITH_ZIP_FOPS = OFF
 LWS_AVOID_SIGPIPE_IGN = OFF
 LWS_WITH_STATS = OFF
 LWS_WITH_SOCKS5 = ON
 LWS_HAVE_SYS_CAPABILITY_H = 
 LWS_HAVE_LIBCAP = 
 LWS_WITH_PEER_LIMITS = OFF
```

**Comment by underhood** [comment]
@stelfrag was caused by typo

**Review by stelfrag — APPROVED** [review]

**Review by mfundul — APPROVED** [review]

**Pull request merged**